### ISSUE TITLE
Wrong execution time in serenity report when features are run in parallel

### ISSUE THREAD

**Issue by preetimohan** [serenity-bdd/serenity-core#2377]
Hi Everyone,

My automation suite uses Cucumber with Java and is integrated with Serenity report. And I use Cucable Plugin (https://github.com/trivago/cucable-plugin) for Feature level parallel execution. However in serenity report Summary page the execution time is not the time elapsed between start of execution and end of execution but it is rather a sum of execution time of all the features, which is wrong in case of parallel execution. Also other values like Fastest test and all are also incorrect in case of parallel execution. 
Can anyone please help debug this issue?

Thanks a lot

**Comment by wakaleo** [comment]
This is the intended behaviour: execution time is the sum of all the execution times, and clock time is the time from start to finish.

I think the fastest and slowest times are taken directly from the scenario durations - what makes you think these are incorrect?

**Comment by preetimohan** [comment]
Even clock time is coming wrong. For Jenkins build execution time  15 mins, following are the serenity reports details:

Execution Time | Clock Time | Fastest Test | Slowest Test | Average Execution Time
-- | -- | -- | -- | --
49m 28s | 5s | 4m 58s | 14m 12s | 9m 53s

This is the case of 5 features executed in parallel. The duration of testcases are 7m 9s, 14m 16s, 5m 3s, 12m 11s, 10m 50s.

No idea why the clock time is coming as 5s

**Comment by wakaleo** [comment]
Have a look at the json test outputs and see if you can see anything unusual there.

**Comment by preetimohan** [comment]
Hi ,

I would like to how the clock time is calculated from the json. Can you please help with same? There is a json corresponding to each feature file and it has a start time and duration as below.
<img width="646" alt="Screenshot 2021-02-10 at 11 11 34 AM" src="https://user-images.githubusercontent.com/17468537/107470680-c9807100-6b91-11eb-8f79-69d13b9c1906.png">

**Comment by GianniGiglio** [comment]
I can confirm I have the same issue when running scenario's in parallel using the --threads option for cucumber. I run my tests using a main class as a fatjar.

**Comment by preetimohan** [comment]
Hi ,

I would like to know how the clock time is calculated from the json. Can you please help with same? There is a json corresponding to each feature file and it has a start time and duration as the screen shot shown above

**Comment by globalworming** [comment]
This is  an open source project and everyone can just have a look and propose changes. @preetimohan, @GianniGiglio, @PavanGurram-DevOps, if this report is useful for you but has a bug, could you maybe get together, maybe find some other java experts, and have a look at the source code. 

You can probably start debugging fairly simple, run one test, have a look at the generated `target/site/serenity/[...].json` file. Check if the runtimes etc seem to make sense, then run `mvn serenity:aggregate` and check the generated report if that makes sense.
If you already see a bug there, you have this json file as example others can try out and you can check out the serenity report module, have a look at the tests, put in breakpoints etc. Together and with the help of colleagues you can probably figure out where the calculation goes wrong and propose changes by creating a Pull Request.

If you see no bug with having a single result json, run two tests in parallel, check the json, aggregate, check report... etc. same as above.

Or hire some short term contractor with the single task "go fix this open source report" :)

**Comment by ankush2704** [comment]
hello @wakaleo in which serenity version this issue got fixed? can you backport it to some 2.x version if not already done. Will be helpful.

**Comment by wakaleo** [comment]
Have a look at 3.1.5.

**Comment by GianniGiglio** [comment]
To clarify: 
In my case the timings where reported wrongly since I as using a custom serenity-runner that would run the tests in parallel on scenario level instead of feature file level. Only then the timings are reported wrongly. 
Since running scenario's in parallel isn't supported by serenity I didn't investigate further.

**Comment by wakaleo** [comment]
> To clarify: In my case the timings where reported wrongly since I was using a custom serenity-runner that would run the tests in parallel on scenario level instead of feature file level. Only then the timings are reported wrongly. Since running scenario's in parallel isn't supported by serenity I didn't investigate further.

Indeed, if you are using a custom runner there is no guarantee that the reporting will be accurate.

**Comment by wakaleo** [comment]
Closing this as the timing reporting has been totally redone in 3.1.x. If you have any further issues in 3.1.x please create a new issue and/or raise a PR with a suggested fix.
